### ISSUE TITLE
[Bug]: (in documentation) xattr "option -r not recognized" on macOS Sonoma 14.7

### ISSUE THREAD

**Issue by peterkaminski** [th-ch/youtube-music#2583]
### Preflight Checklist

- [X] I use the latest version of YouTube Music (Application).
- [X] I have searched the [issue tracker](https://github.com/th-ch/youtube-music/issues) for a bug report that matches the one I want to file, without success.
- [X] I understand that **th-ch/youtube-music has NO affiliation with Google or YouTube**

### YouTube Music (Application) Version

3.6.2

### Checklists

- [ ] I use the portable version of the YouTube Music Application.
- [ ] I can reproduce this issue in the [official version of (WEB) YTM](https://music.youtube.com).

### What operating system are you using?

macOS

### Operating System Version

macOS Sonoma Version 14.7 (Apple M1 Max)

### What CPU architecture are you using?

arm64 (including Apple Silicon)

### Reproduction steps

1. Read the [macOS section in README.md](https://github.com/th-ch/youtube-music/?tab=readme-ov-file#macos)
2. Click the copy icon at the right of the `xattr` command.
3. Clipboard now has "xattr -cr /Applications/YouTube\ Music.app"
4. Paste into terminal window.

### Expected Behavior

`xattr` command completes with no errors, and newly-downloaded YouTube Music app runs.

### Actual Behavior

`xattr` outputs error message "option -r not recognized" and usage notice, and does not proceed.

Command:

```shell
xattr -cr /Applications/YouTube\ Music.app
```

Output:

```
option -r not recognized

usage: xattr [-slz] file [file ...]
       xattr -p [-slz] attr_name file [file ...]
       xattr -w [-sz] attr_name attr_value file [file ...]
       xattr -d [-s] attr_name file [file ...]
       xattr -c [-s] file [file ...]

The first form lists the names of all xattrs on the given file(s).
The second form (-p) prints the value of the xattr attr_name.
The third form (-w) sets the value of the xattr attr_name to attr_value.
The fourth form (-d) deletes the xattr attr_name.
The fifth form (-c) deletes (clears) all xattrs.

options:
  -h: print this help
  -s: act on symbolic links themselves rather than their targets
  -l: print long format (attr_name: attr_value)
  -z: compress or decompress (if compressed) attribute value in zip format
```

### Enabled plugins

none

### Additional Information

I used `xattr -c /Applications/YouTube\ Music.app` and the newly-downloaded YouTube Music app worked. (I did not test if it worked before running the `xattr` command.)

I don't know what the "-r" flag was supposed to mean, but perhaps it's okay to just change the README page to "-c"?  Should be verified to be okay before someone makes the change.

`xattr` doesn't have its own version number, I believe it's tied to the macOS version number.

Copying and pasting the command from the README.md used to work for me, but now it doesn't. I don't know the details of the working command I'd used before.

**Comment by ArjixWasTaken** [comment]
`-r` stands for `remove`
`-c` worked because it stands for `clear`

they are not the same, but it makes sense why both work
I am not sure why `-r` was removed though

**Comment by th-ch** [comment]
👋 That's weird, it works here on MacOS 14.7.1 (from running `sw_vers`) - running `xattr --help` gives me
```
The fifth form (-c) deletes (clears) all xattrs.

options:
  -h: print this help
  -l: print long format (attr_name: attr_value and hex output has offsets and
      ascii representation)
  -r: act recursively
  -s: act on the symbolic link itself rather than what the link points to
  -v: also print filename (automatic with -r and with multiple files)
  -x: attr_value is represented as a hex string for input and output
```
(`which xattr` gives me `/usr/bin/xattr`)

**Comment by peterkaminski** [comment]
D'oh, I did not think of trying `which`, and I should have.

`sw_vers` gives me macOS 14.7, build version 23H124.

`which xattr` gives me `/opt/homebrew/bin/xattr`

`/usr/bin/xattr -h` shows that it has the `-r` option.

So, my experience with `xattr` is due to it being from Homebrew rather than using the built-in `xattr`. That makes this not a bug report, but perhaps **an enhancement request, to either remove the `-r` from the instructions, or to specify the full command path, `/usr/bin/xattr` (or both).**

I tried to look for why Homebrew installed an `xattr`; it doesn't show up in `brew list`, nor with `brew list --formula | xargs -I {} sh -c 'brew list {} | grep -q "xattr" && echo {}'`.

Perhaps it's just installed with Homebrew itself?

Here's another project's fix: <https://gogs.hasufell.de/hasufell/ghcup-hs/commit/c914a284ded419f126d279cf7556f9030b50f2a4>

For background, the Homebrew version of `xattr` is a wrapper around the [Python `xattr` library](https://github.com/xattr/xattr), maybe they include their own for Linux compatibility?

**Comment by peterkaminski** [comment]
tl;dr: I propose to change this from a bug report to an enhancement request:

- remove the `-r` from the instructions (I don't understand the implications, though), or
- specify the full command path, `/usr/bin/xattr` (seems safe and practical)
- or both of the above (if removing `-r` is okay)

**Comment by ArjixWasTaken** [comment]
My question is why would they make a wrapper that is not compatible with the original xattr?

**Comment by peterkaminski** [comment]
@ArjixWasTaken, the wrapper is generally compatible, it just doesn't have the `-r` option which has been added to the Apple version.

There is an issue and proposal dated 2024-04-03 to add `-r` to the Python library: <https://github.com/xattr/xattr/issues/118>

(This comment has no bearing on my proposal above.)

**Comment by peterkaminski** [comment]
I believe PR #2586 would close this issue.

A dangling question might be whether `-r` is needed, but it doesn't seem too important to resolve, since this PR would eliminate the presenting problem.
